src/scripts/tray.js

```diff
@@ -1,22 +1,48 @@
-const { Tray } = require("electron");
-const { getMenu } = require("./menu");
+const { Tray, app } = require("electron");
+const { Menu } = require("electron");
+const { getMenu, mainMenu } = require("./menu");
+const { store, settings } = require("./settings");
 const trayModule = {};
 let tray;
 
 trayModule.addTray = function (options = { icon: "" }) {
   tray = new Tray(options.icon);
 };
 
-trayModule.refreshTray = function () {
+trayModule.refreshTray = function (mainWindow) {
   if (!tray) {
     trayModule.addTray();
   }
+
   tray.on("click", function (e) {
-    // do nothing on click
+    if (mainWindow) {
+      mainWindow.show();
+    }
   });
 
   tray.setToolTip("Tidal-hifi");
-  tray.setContextMenu(getMenu());
+  
+  if (mainWindow && store.get(settings.minimizeOnClose)) {
```

Comment:
We lose the entire menu like this, we should do either one of two solutions:

- Add the items to the menu regardless of the setting and just use `getMenu()`
- Merge the result of `getMenu()` with this new menu.

Either way is fine for me. Being able to focus (show) and close the app from the menu is nice anyway.

Reply:
Yeah, I did that (I left a https://github.com/Mastermindzh/tidal-hifi/issues/59#issuecomment-862293724 on the issue thread) but I didn't know if you were about to agree with it. If the comment looks OK I'll implement it